hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFormatReader.java

```diff
@@ -172,21 +171,35 @@ public boolean hasNext() {
     }
   }
 
-  private boolean readMagic() throws IOException {
+  // TODO : separate reading of markers once idempotency is resolved.
+  private boolean readMarkers() throws IOException {
     try {
-      // 1. Read magic header from the start of the block
+      // 1. Read end marker from end of log file or 2. Read magic header from the start of the block
       inputStream.readFully(magicBuffer, 0, 4);
-      if (!Arrays.equals(magicBuffer, HoodieLogFormat.MAGIC)) {
-        throw new CorruptedLogFileException(
-            logFile + "could not be read. Did not find the magic bytes at the start of the block");
-      }
-      return false;
+      if(checkEndMarker())
+        return true;
+      return checkMagic();
     } catch (EOFException e) {
       // We have reached the EOF
       return true;
     }
   }
 
+  private boolean checkEndMarker() {
+    if (Arrays.equals(magicBuffer, HoodieLogFormat.END_MARKER)) {
```

Comment:
probably want to rename `magicBuffer` to `markerBuffer` and the methods to `isEndMarker()`  and `isMagicMarker`?

Reply:
unrelated to this PR, but should `magicBuffer` be a static byte[] at all? I def dont think it should be static (appreciate it if you can change this) and think we should just have it allocated locally, as opposed to a member variable that methods like checkEndMarker implicitly access.